src/visualization/utils/useStaticLegend.ts

```diff
@@ -73,6 +73,64 @@ export const useStaticLegend = (properties): StaticLegendConfig => {
 
   const {isViewingVisOptions} = useSelector(getActiveTimeMachine)
   const dispatch = useDispatch()
+
+  const {
+    legendColorizeRows = LEGEND_COLORIZE_ROWS_DEFAULT,
+    legendOpacity = LEGEND_OPACITY_DEFAULT,
+    legendOrientationThreshold = LEGEND_ORIENTATION_THRESHOLD_DEFAULT,
+  } = properties
+
+  const {
+    staticLegend = {
+      colorizeRows: LEGEND_COLORIZE_ROWS_DEFAULT,
+      heightRatio: STATIC_LEGEND_HEIGHT_RATIO_NOT_SET,
+      opacity: LEGEND_OPACITY_DEFAULT,
+      orientationThreshold: LEGEND_ORIENTATION_THRESHOLD_DEFAULT,
+      show: STATIC_LEGEND_SHOW_DEFAULT,
+      widthRatio: STATIC_LEGEND_WIDTH_RATIO_DEFAULT,
+    } as StaticLegendAPI,
+  } = properties
+
+  const {
+    colorizeRows = false, // undefined is false because of omitempty
+    heightRatio = STATIC_LEGEND_HEIGHT_RATIO_NOT_SET,
+    opacity = LEGEND_OPACITY_DEFAULT,
+    orientationThreshold = legendOrientationThreshold,
+    show = STATIC_LEGEND_SHOW_DEFAULT,
+    ...config
+  } = staticLegend
+
+  useEffect(() => {
+    const shouldStaticLegendPropertiesBeUpdated: boolean =
+      isViewingVisOptions &&
+      !show &&
+      heightRatio === STATIC_LEGEND_HEIGHT_RATIO_NOT_SET
+
+    if (!shouldStaticLegendPropertiesBeUpdated) {
+      return
+    }
+    const validOpacity =
+      typeof legendOpacity === 'number' &&
+      legendOpacity === legendOpacity &&
+      legendOpacity >= LEGEND_OPACITY_MINIMUM &&
+      legendOpacity <= LEGEND_OPACITY_MAXIMUM
+        ? legendOpacity
+        : LEGEND_OPACITY_DEFAULT
+
+    const validThreshold =
+      typeof legendOrientationThreshold !== 'number' ||
+      legendOrientationThreshold !== legendOrientationThreshold ||
+      legendOrientationThreshold > 0
+        ? LEGEND_ORIENTATION_THRESHOLD_HORIZONTAL
+        : LEGEND_ORIENTATION_THRESHOLD_VERTICAL
+
+    updateStaticLegendProperties({
+      colorizeRows: legendColorizeRows,
+      opacity: validOpacity,
+      orientationThreshold: validThreshold,
+    })
+  }, [isViewingVisOptions, show, heightRatio])
```

Comment:
This is out of order. It looks like `updateStaticLegendProperties` is defined later.

Reply:
No need to address this, just wanted to comment that your proposed fix had things out of order.

Reply:
No need to address this, just wanted to comment that your proposed fix had things out of order.